diff --git a/README.md b/README.md
index 3920c3f..23dca8c 100644
--- a/README.md
+++ b/README.md
@@ -43,6 +43,8 @@ Special reward system: Random block rewards
 
 ### Wow plz make dogecoind
 
+#### Linux:
+
     sudo apt-get install build-essential \
                          libssl-dev \
                          libdb5.1++-dev \
@@ -53,6 +55,14 @@ Special reward system: Random block rewards
     cd src/
     make -f makefile.unix USE_UPNP=1 USE_IPV6=1 USE_QRCODE=1
 
+#### Mac OSX:
+
+    brew install boost miniupnpc openssl berkeley-db
+    openssl version # should be OpenSSL 1.0.1e
+                    # if not, then do `brew link openssl --force`
+    cd src/
+    make -f makefile.osx USE_UPNP=1 USE_IPV6=1 USE_QRCODE=1
+
 ### Such ports
 RPC 22555
 P2P 22556
diff --git a/src/makefile.osx b/src/makefile.osx
index 1c56866..69a967d 100644
--- a/src/makefile.osx
+++ b/src/makefile.osx
@@ -7,17 +7,15 @@
 # Originally by Laszlo Hanyecz (solar@heliacal.net)
 
 CXX=llvm-g++
-DEPSDIR=/opt/local
+DEPSDIR=/usr/local
 
 INCLUDEPATHS= \
  -I"$(CURDIR)" \
  -I"$(CURDIR)"/obj \
- -I"$(DEPSDIR)/include" \
- -I"$(DEPSDIR)/include/db48"
+ -I"$(DEPSDIR)/include"
 
 LIBPATHS= \
- -L"$(DEPSDIR)/lib" \
- -L"$(DEPSDIR)/lib/db48"
+ -L"$(DEPSDIR)/lib"
 
 USE_UPNP:=1
 
@@ -30,7 +28,7 @@ ifdef STATIC
 TESTLIBS += \
  $(DEPSDIR)/lib/libboost_unit_test_framework-mt.a
 LIBS += \
- $(DEPSDIR)/lib/db48/libdb_cxx-4.8.a \
+ $(DEPSDIR)/lib/libdb_cxx.a \
  $(DEPSDIR)/lib/libboost_system-mt.a \
  $(DEPSDIR)/lib/libboost_filesystem-mt.a \
  $(DEPSDIR)/lib/libboost_program_options-mt.a \
@@ -42,7 +40,7 @@ else
 TESTLIBS += \
  -lboost_unit_test_framework-mt
 LIBS += \
- -ldb_cxx-4.8 \
+ -ldb_cxx \
  -lboost_system-mt \
  -lboost_filesystem-mt \
  -lboost_program_options-mt \
